sql/catalyst/src/main/java/org/apache/spark/sql/BaseRow.java

```diff
@@ -155,27 +155,6 @@ public int fieldIndex(String name) {
     throw new UnsupportedOperationException();
   }
 
-  /**
-   * A generic version of Row.equals(Row), which is used for tests.
-   */
-  @Override
```

Comment:
Existing: can you add some javadoc to this class to explain what its used for and why its in Java?